packages/react/src/components/UIShell/__tests__/SideNavMenu-test.js

```diff
@@ -98,4 +102,118 @@ describe('SideNavMenu', () => {
     );
     expect(ref).toHaveBeenCalledWith(screen.getByRole('button'));
   });
+
+  it('should return true if the child is a valid React element, and instance of Array and has isActive and aria-current props', () => {
+    const child = [
+      <SideNavMenuItem isActive={true} aria-current="page">
+        <SideNavMenuItem isActive={false} aria-current="page">
+          a
+        </SideNavMenuItem>
+      </SideNavMenuItem>,
+      <SideNavMenuItem isActive={true} aria-current="page">
+        b
+      </SideNavMenuItem>,
+    ];
+    expect(hasActiveDescendant(child)).toBe(true);
+  });
+
+  it('should return true if the child is a invalid React element and has isActive props set to true', () => {
+    const child = <div isActive={true} />;
+    expect(hasActiveDescendant(child)).toBe(true);
+  });
```

Comment:
Same kind of idea here. I don't think we need to directly assert that the function call returns true. Instead it's the end result of the function call we want to ensure happens correctly.


Again I think in nearly all cases this would be an assertion of `toHaveClass` or `.not.toHaveClass`. The classes are admittedly still an implementation detail, but they're the closest thing to what the user perceives without having to actually assert that certain styles are applied (background, etc. or whatever is in selectors with `${prefix}--side-nav__item--active`